tests/test_history.py

```diff
@@ -557,6 +558,277 @@ def handle_data(context, data):
         self.assertEquals(139.36946942498648, last_prices[oldest_dt])
         self.assertEquals(180.15661995395106, last_prices[newest_dt])
 
+    def test_history_in_bts_price_days(self):
+        """
+        Test calling history() in before_trading_start()
+        with daily price bars.
+        """
+        algo_text = """
+from zipline.api import history
+
+def initialize(context):
+    context.first_day = True
+
+def before_trading_start(context, data):
+    if not context.first_day:
+        prices_bts = history(bar_count=3, frequency='1d', field='price')
+        context.prices_bts = prices_bts
+
+def handle_data(context, data):
+    prices_hd = history(bar_count=3, frequency='1d', field='price')
+    context.prices_hd = prices_hd
+    context.first_day = False
+""".strip()
+
+        #      March 2006
+        # Su Mo Tu We Th Fr Sa
+        #          1  2  3  4
+        #  5  6  7  8  9 10 11
+        # 12 13 14 15 16 17 18
+        # 19 20 21 22 23 24 25
+        # 26 27 28 29 30 31
+        start = pd.Timestamp('2006-03-20', tz='UTC')
+        end = pd.Timestamp('2006-03-22', tz='UTC')
+
+        sim_params = factory.create_simulation_parameters(
+            start=start, end=end)
+
+        test_algo = TradingAlgorithm(
+            script=algo_text,
+            data_frequency='minute',
```

Comment:
Updated so that those two tests which call daily `history()` are run in both minutely an daily mode. I've not run the minutely `history()` calls in daily mode as this raises an `ncompatibleHistoryFrequency` exception.